test/e2e/storage/framework/testdriver.go

```diff
@@ -185,6 +185,8 @@ const (
 	// capacity information for it.
 	CapCapacity Capability = "capacity"
 
+	// anti-capability for drivers that do not support filesystem resizing
```

Comment:
Is this specifically referring to the behavior of expanding during NodeStage?

How does this interact with `CapControllerExpansion` and `CapNodeExpansion`?

Reply:
> Is this specifically referring to the behavior of expanding during NodeStage?

Not really. A CSI driver could do the resize elsewhere even though it should probably be in `NodeStage`. The e2e test I'm adding does not care how/wherethe driver does the resize, it just restores from a snapshot and checks the size: https://github.com/kubernetes/kubernetes/pull/110304/files#diff-c0b41d6b4f963df4d44f35004220c427e991fa0606ec6bdc3ffbeef38e27bba6R470

> How does this interact with `CapControllerExpansion` and `CapNodeExpansion`?

There is no interaction in the code now. Those capabilities indicate if a driver can expand a volume - the underlying block device size. But that does not necessarily mean it resizes he filesystem correctly when restoring from a snapshot or clone. 

So the "interaction" between the capabilities basically is that `CapControllerExpansion` and `CapNodeExpansion` are prerequisites for filesystem resizing. So if there is a driver that runs this e2e and *can* expand volumes, but can *not* resize filesystem properly, it will need to set the newly added anti-capability to true (`CapFilesystemResizeNotSupported`).

The issue originated here: https://github.com/kubernetes/kubernetes/issues/94929

Reply:
Thanks, I was confused because when we normally talk about FS resize, we generally refer to the NodeExpansion feature.

At the risk of being too verbose, what if we name the field to `CapFSResizeFromSourceNotSupported`?